### DISCUSSION TITLE
Using a TableSeeder

### DISCUSSION THREAD

**Discussion by rmclain** [Z3d0X/filament-fabricator#128] [Q&A] [answered]
I'm trying to build pages in staging, and then post those pages to production using a PagesTableSeeder, but even if I copy the table structure and SQL inserts for the seeder, I get errors when trying to load the /pages/ url. It gets stuck at the invalid index #. Is there a way to create a PagesTableSeeder that will allow this type of functionality?

**Comment by mybouhssina** [comment, marked as answer]
Hey, did you try to run php artisan cache:clear after creating the pages ?
I something similar: created manually a page in the database pages table and open the 'pages' page in Filament admin panel and got the a similar invalid index error, then I cleared the cache and it s now working properly.

**Reply by rmclain** [reply]
Thank you for the tip; I will certainly keep it in mind. It started working after I switched from using a raw DB query in my seeder to directly using the model's create function (Z3d0X\FilamentFabricator\Models\Page::create). Once I ran the seeder with this code, everything seemed to just work.

```php
use Z3d0X\FilamentFabricator\Models\Page;
use Illuminate\Database\Seeder;

class PagesTableSeeder extends Seeder
{
    public function run()
    {
        $navigation = '{"type":"navigation","data":{"menuItems":{"Features":"/#features","Integrations":"/#integrations","How Does It Work?":"/#howdoesitwork","Plans & Pricing":"/#pricing"},"cta":{"Get Started":"/#pricing"}}}';
        $headerBanner = '{"type":"call-to-action-left","data":{"....."}}';
        $blocks = "[{$navigation},{$headerBanner}]";

        Page::create([
            'id' => 1,
            'title' => 'Home',
            'slug' => '/',
            'layout' => 'marketing',
            'blocks' => json_decode($blocks, true),
            'parent_id' => null,
            'created_at' => '2024-01-17 16:41:00',
            'updated_at' => '2024-01-17 17:23:49'
        ]);
    }
}
```

**Reply by mybouhssina** [reply]
that makes sense, since there is an event listener in the default Page Model that takes care of deleting the cache:
![image](https://github.com/Z3d0X/filament-fabricator/assets/18598565/bbabf6df-d453-4c34-a808-5d21e34ca9b4)
